code/modules/mob/mob_movement.dm

```diff
@@ -1,5 +1,5 @@
 /mob/CanPass(atom/movable/mover, turf/target, height=1.5, air_group = 0)
-	if(air_group || (height==0)) return 1
+	if(air_group || (height==0) || (incorporeal_move == INCORPOREAL_ETHEREAL)) return 1

```

Comment:
-wizard gets knocked down, his density becomes 0
-he uses etheral jaunt, his density is still 0
-while etheral he gets up, his density becomes 1
-he gets hit by a laser while still etheral
blargharbl